### ISSUE TITLE
Form - Ensure custom controls provide proper textual name, role, and state information - (2036570495)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5629]
### Module: 
 40 Form 

### Violation: 
 Ensure custom controls provide proper textual name, role, and state information 

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570495 

### Severity: 
 7 

### Description: 
 [Issue]
There is info control without appropriate name, role, and/or state information.

[User Impact]
Screen reader users will have difficulty determining the purpose and state of these controls.

[Code Reference] 
 ```
<calcite-icon tabindex="0" id="theme-help" icon="question" scale="s" aria-describedby="calcite-tooltip-cb59b258-d075-178c-1434-738003cc025b" aria-hidden="true" calcite-hydrated=""></calcite-icon>
 ``` 

### Note: 
 [Suggestion]
Ensure custom controls provide proper textual name, role, and state information. Properly structured native controls provide this information automatically. For custom controls, developers must explicitly add this information by using ARIA attributes.

Most modern browsers have an Accessibility section in the developer tools panel that will display an element&#039;s calculated name, role, and state information.

A control&#039;s name must represent the general purpose of the control and include any visible labeling text. It can be set through a variety of methods. Custom controls typically use the aria-label attribute or the aria-labelledby attribute to supply the name.

A control&#039;s role indicates the type of control, such as "checkbox" or "button". For custom controls, the role must be set with the role attribute. The ARIA specification only permits certain values for this attribute. Additionally, the ARIA in HTML specification does prevents the use of certain roles on some elements. For a list of HTML elements and their permitted roles, see the ARIA in HTML specification: https://www.w3.org/TR/html-aria/#docconformance

A control&#039;s state indicates several possible pieces of information. Common states include expanded, current, selected, checked, and disabled. Most are indicated with similarly named ARIA attributes. Certain states can only be used with certain roles. For a list of roles and their allowed states, see the ARIA specification: https://www.w3.org/TR/wai-aria/#role_definitions

The Web Accessibility Initiative (WAI) of the W3C publishes suggested patterns for custom controls in the WAI-ARIA Authoring Practices: https://www.w3.org/TR/wai-aria-practices/ These patterns are not tested or endorsed by the W3C, but they often serve as good starting points for developing custom controls.

[Compliant Code Example] 
 ```
<calcite-icon role="button" aria-label="your name info" tabindex="0" id="theme-help" icon="question" scale="s" aria-describedby="calcite-tooltip-cb59b258-d075-178c-1434-738003cc025b" calcite-hydrated=""></calcite-icon>
 ```
 

### Media Type: 
 Custom Controls 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Mitigated with the following [Codepen](https://codepen.io/geospatialem/pen/YzvEXBo) using the `icon`'s `textLabel` prop and surrounding the `tooltip` content in a `span` element.
